### PR TITLE
Mob-ifys Bats & Dr. Acula

### PR DESCRIPTION
**Pull request by Glamurio**
[Feature] [Critters]

## About the PR 
Ports all of the features from `/obj/critter/bat` to `/mob/living/critter/small_animal/bat`.

Adds a `bat.dm` alongside `/datum/aiHolder/bat` with all of the bat AI handling.

Adds a new ability `/datum/targetable/critter/drink_blood` which bats use to drink blood.

Replaces all `/obj/critter/bat/doctor` with `/mob/living/critter/small_animal/bat/doctor`, and changes the handling for the Dr. Acula crew objectives.

There may be a slight increase in bats trying to drink blood.

## Why's this needed? 
Object critter bad. Also Dr. Acula needs to fit inside a pet carrier.

## Changelog 

```changelog
(u)Glamurio (Ryou)
(+)Dr. Acula now fits inside of pet carriers.
```

### CODE REVIEW

File: code/datums/abilities/critter/small_animal.dm [hunk 1]
```diff
@@ -283,3 +283,146 @@
...
+	cast(atom/target)
+		if (..())
+			return TRUE
+		if (target == holder.owner)
+			return TRUE
+		if (!isliving(holder.owner))
+			return TRUE
+
+		var/mob/living/M = holder.owner
+		var/datum/abilityHolder/critter/C = holder
```
Comment (TobleroneSwordfish) [inline code/datums/abilities/critter/small_animal.dm:305]: pls don't use non standard single letter variable names like this

File: code/datums/abilities/critter/small_animal.dm [hunk 2]
```diff
@@ -283,3 +283,146 @@
...
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_STUNNED
+	icon = 'icons/ui/actions.dmi'
+	icon_state = "blood"
+	bar_icon_state = "bar-vampire"
+	border_icon_state = "border-vampire"
+	color_active = "#d73715"
+	color_success = "#f21b1b"
+	color_failure = "#8d1422"
+	var/mob/living/critter/small_animal/bat/mini_vampire
+	var/datum/abilityHolder/critter/C
```
Comment (TobleroneSwordfish) [inline code/datums/abilities/critter/small_animal.dm:322]: Ditto

File: code/mob/living/critter/small_animal.dm [hunk 3]
```diff
@@ -2957,6 +2965,56 @@ var/list/mob_bird_species = list("smallowl" = /mob/living/critter/small_animal/b
...
+
+	// in here we'll search for reagent containers with blood
+	seek_target(var/range = 7)
+		. = list()
+
+		for (var/obj/fluid/F in view(range, src))
+			if (F.name == "blood")
+				. += F
+
+		for (var/obj/item/reagent_containers/S in view(range, src))
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/small_animal.dm:3006]: S?!?!!!

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Bugs found from some basic testing:
Dr. Acula only drinks once before becoming satisfied, suffers hypnotension really really fast and eventually bloodgibs if you feed them too much.

Consider just giving Acula a vampire ability holder and using the standard vampire drink ability rather than trying to recreate parts of it here?

File: code/mob/living/critter/small_animal.dm (hunk 3 above)
Comment (Glamurio) [reply code/mob/living/critter/small_animal.dm:3006]: This was all copied-over from previous AI, I did not question its insanity.

**Pull request closed without merging**